File: src/main/java/io/chesslave/hands/Mover.java
package io.chesslave.hands;

import io.chesslave.model.Square;

/**
 * An entity able to move pieces on the board.
 */
public interface Mover {

    /**
     * Move a piece.
     *
     * @param from starting square
     * @param to destination square
     */
    void move(Square from, Square to);
}

File: backend/src/main/java/io/chesslave/hands/Mover.kt
package io.chesslave.hands

import io.chesslave.model.Square

/**
 * An entity able to move pieces on the board.
 */
interface Mover {

    /**
     * Move a piece.
     *
     * @param from starting square
     * @param to destination square
     * @throws MoverException if something goes wrong executing the move
     */
    fun move(from: Square, to: Square)
}